esp/services/ws_workunits/ws_workunitsHelpers.hpp

```diff
@@ -260,7 +260,7 @@ struct WUComponentLogOptions
         }
 
         if (componentsFilterObj != nullptr)
-            logFetchFilter = getBinaryLogAccessFilter(logFetchFilter, componentsFilterObj, LOGACCESS_FILTER_and);
+            logFetchFilter.setown(getBinaryLogAccessFilterOwn(logFetchFilter.getLink(), componentsFilterObj, LOGACCESS_FILTER_and));
```

Comment:
per offline convo: let's try to externalize the repeating logic into a function.